core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -1286,13 +1289,31 @@ public Result visit(Uncollect e) {
 
   public Result visit(TableFunctionScan e) {
     final List<SqlNode> inputSqlNodes = new ArrayList<>();
+    final List<SqlNode> fieldNodes = new ArrayList<>();
+
+    final int filedCount = e.getRowType().getFieldCount();
```

Comment:
sorry, didn't notice this before, but I think this should be "fieldCount" instead of "filedCount"